### PR TITLE
initramfs: fix busybox 1.36.1 build witk kernel 6.8

### PR DESCRIPTION
**Pull request by sbraz**
Patch taken from
https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=d8ad860a1ed9aa92ada.

Bug: https://bugs.gentoo.org/928573

### CODE REVIEW

**Comment by steils** [comment]
@robbat2 @ZeroChaos-

**Comment by WLMOSV** [comment]
Does this apply to my problem?

**Comment by sbraz** [comment]
> Does this apply to my problem?

Yes, please read https://bugs.gentoo.org/928573

**Comment by WLMOSV** [comment]
Thank you very much for your patch. I was depressed for several days and finally solved it!

**Comment by WLMOSV** [comment]
hi I added the script to my directory and it looks like this, but the compilation still doesn’t pass.

**Comment by WLMOSV** [comment]
solved

File: patches/busybox/1.36.1/busybox-1.36.1-kernel-6.8.patch [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+--- a/networking/tc.c
++++ b/networking/tc.c
+@@ -231,6 +231,13 @@ static int cbq_parse_opt(int argc, char **argv, struct nlmsghdr *n)
+ 	return 0;
+ }
+ #endif
++
++#ifndef TCA_CBQ_MAX
++/*
++ * Linux v6.8-rc1~131^2~60^2^2 removed the uapi definitions for CBQ.
```
Comment (Jamim) [inline patches/busybox/1.36.1/busybox-1.36.1-kernel-6.8.patch]: It looks like this line is corrupted.
```suggestion
+ * Linux v6.8-rc1 removed the uapi definitions for CBQ.
```

File: patches/busybox/1.36.1/busybox-1.36.1-kernel-6.8.patch [hunk 2]
```diff
@@ -0,0 +1,53 @@
+http://lists.busybox.net/pipermail/busybox/2024-March/090678.html
+https://bugs.gentoo.org/926872
```
Comment (Jamim) [inline patches/busybox/1.36.1/busybox-1.36.1-kernel-6.8.patch]: I believe it's worth using a direct link to the corresponding bug.
```suggestion
https://bugs.gentoo.org/928573
```

**Review by Jamim — CHANGES_REQUESTED** [review]
_Hello @sbraz,_

Thank you for the fix!
There are just a couple of minor suggestions.

_Best regards!_

File: patches/busybox/1.36.1/busybox-1.36.1-kernel-6.8.patch (hunk 1 above)
Comment (sbraz) [reply patches/busybox/1.36.1/busybox-1.36.1-kernel-6.8.patch]: Yeah you're right, I didn't really try to fix all that from the patch in the ML. Done.

**Review by Jamim — APPROVED** [review]

**Comment by Jamim** [comment]
_Hello @ceamac,_

Would you mind helping us with this PR?
It's based on your changes from gentoo/gentoo@d8ad860a1ed9aa92adaa7dcf1c3fc78d0e2f80ce.

_Thanks in advance!_

**Pull request closed without merging**